core/services/relay/dydx/config_digester.go

```diff
@@ -0,0 +1,48 @@
+package dydx
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+const ConfigDigestPrefixDydx types.ConfigDigestPrefix = 4
+
+var _ types.OffchainConfigDigester = (*OffchainConfigDigester)(nil)
+
+type OffchainConfigDigester struct {
+	// endpoint type can be staging or prod
+	endpointType string
+}
+
+func (d OffchainConfigDigester) ConfigDigest(cfg types.ContractConfig) (types.ConfigDigest, error) {
+	return d.configDigest()
+}
+
+// The digest is unique per OffchainConfigDigester.endpointType value. This ensures
+// protocol instances for staging vs prod are distinct, and we have separate
+// monitoring for each.
+func (d OffchainConfigDigester) configDigest() (types.ConfigDigest, error) {
+	digest := types.ConfigDigest{}
+	buf := sha256.New()
+
+	if err := binary.Write(buf, binary.BigEndian, uint8(len(d.endpointType))); err != nil {
```

Comment:
This could probably done in a more collision-resistant manner 😉

Reply:
Yes, sorry about that. It was just initially trying to get the structure right. Fixed now.